src/weapon/w_050.c

```diff
@@ -486,7 +486,122 @@ void func_ptr_8017000C(Entity* self) {
     }
 }
 
-INCLUDE_ASM("weapon/nonmatchings/w_050", func_ptr_80170010);
+s32 func_ptr_80170010(Entity* self) {
+    const int PrimCount = 8;
+    Primitive* prim;
+    Point16 point;
+    s32 params;
+    s32 i;
+    s16 x, y;
+    s16 baseX, baseY;
+    s16 prim_x, prim_y;
+    s32 result;
+
+    if (PLAYER.ext.player.anim < 0x41 || PLAYER.ext.player.anim > 0x47 ||
+        !g_Player.unk46) {
+        DestroyEntity(self);
+        return;
+    }
+
+#ifdef VERSION_PSP
+    params = (self->params & 0x7F00) >> 8;
```

Comment:
psp version matches on ps1 from what I see

Reply:
I must have split these when other mismatched caused some divergence. After I had a match on both sides I never went back to check if they still differed.